R/Connection.R

```diff
@@ -645,18 +703,89 @@ odbcListDrivers <- function(keep = getOption("odbc.drivers_keep"), filter = getO
   res
 }
 
-#' List Available Data Source Names
+#' List Configured Data Source Names
 #'
-#' List the available data sources on your system. See the [DSN Configuration
-#' files](https://github.com/r-dbi/odbc#dsn-configuration-files) section of the
-#' package README for details on how to install data sources for the most
-#' common databases.
+#' @description
 #'
-#' @return A data frame with two columns.
+#' Collect information about the available data source names (DSNs). A DSN must
+#' be both installed and configured with the driver manager to be included in
+#' this list. Configuring a DSN just sets up a lookup table to allow users to
+#' pass only the DSN to [dbConnect()], which will then retrieve the
+#' data source information from the driver manager.
+#'
+#' DSNs that are not configured with the driver manager can still be
+#' connected to with [dbConnect()] by providing DSN metadata directly.
+#'
+#' @return A data frame with two columns:
 #' \describe{
-#'   \item{name}{Name of the data source}
-#'   \item{description}{Data Source description}
+#'   \item{name}{Name of the data source. The entries in this column can be
+#'     passed to the `dsn` argument of [dbConnect()].}
```

Comment:
Should check that `[dbConnect()]` links to the right place — want to make sure it links specifically to the docs for the odbc method.

Reply:
Heard! Links to the docs for the odbc method specifically, even with DBI loaded, unless we write [DBI::dbConnect()].